### PR TITLE
Update pilot presubmit to use rbac cluster

### PR DESCRIPTION
**Pull request by sebastienvas**
Remove cluster for code coverage

```release-note
none
```

### CODE REVIEW

File: prow/config.yaml [hunk 1]
```diff
@@ -593,8 +593,6 @@ presubmits:
           - name: pilot-codecov-token
             mountPath: /etc/codecov
             readOnly: true
-          - name: e2e-testing-kubeconfig
```
Comment (yutongz) [inline prow/config.yaml]: You may want to keep this. Pilot needs the existence of this file to pass bazel build before calculate coverage.

File: prow/config.yaml [hunk 2]
```diff
@@ -593,8 +593,6 @@ presubmits:
...
-            mountPath: /home/bootstrap/.kube
```
Comment (yutongz) [inline prow/config.yaml:597]: You may want to keep this. Pilot needs the existence of this file to pass bazel build before calculate coverage.

**Review by yutongz — CHANGES_REQUESTED** [review]

File: prow/config.yaml (hunk 2 above)
Comment (sebastienvas) [reply prow/config.yaml:597]: Code coverage should not need it
https://github.com/istio/pilot/pull/1254

**Review by yutongz — APPROVED** [review]

**Pull request merged**